app/src/main/java/com/wesleyfranks/todo24/data/TodoAdapter.kt

```diff
@@ -82,7 +78,7 @@ class TodoAdapter(private val completedChecked: CompletedChecked, private val cl
     }
 
     interface ClickedTodo{
-        fun OnItemClicked(todo: Todo, pos: Int)
+        fun OnItemClicked(editViewTodo: Todo, pos: Int)
```

Comment:
The interface approach works, but isn't really necessary for a single abstract method (SAM). You can just use a lambda instead. So instead of:
```
class ViewHolder(
        private val itemBinding: TodoListItemBinding,
        private val completedChecked: CompletedChecked,
        private val clickedTodo: ClickedTodo,
        private val deleteTodo: DeleteTodo)
```
You can have something like:
```class ViewHolder(
        private val itemBinding: TodoListItemBinding,
        private val completeChecked: (todo: Todo, pos: Int) -> Unit,
        private val editTodo: (todo: Todo, pos: Int) -> Unit,
        private val deleteTodo: (todo: Todo, pos: Int) -> Unit)
```